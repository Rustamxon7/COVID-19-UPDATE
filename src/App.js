import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import AppContainer from './components/Home/Container';

const App = () => {
  return (
    <div>
      <Router>
        <AppContainer />
      </Router>
    </div>
  );
};

export default App;
