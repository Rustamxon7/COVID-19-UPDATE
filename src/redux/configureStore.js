import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import dataReducer from './covid-19/covidUpdate';
import historyReducer from './covid-19/covidHistory';

const reducer = combineReducers({
  covid: dataReducer,
  history: historyReducer,
});

const store = createStore(reducer, applyMiddleware(thunk, logger));

export default store;
