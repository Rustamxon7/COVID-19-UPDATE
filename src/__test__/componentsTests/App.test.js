import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import dataReducer from '../../redux/covid-19/covidUpdate';
import App from '../../App';

const reducer = combineReducers({
  covid: dataReducer,
});

const store = createStore(reducer, applyMiddleware(logger, thunk));

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
