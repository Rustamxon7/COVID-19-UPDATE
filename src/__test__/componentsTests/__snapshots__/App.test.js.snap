// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <div
    className="app-container"
  >
    <div>
      <div
        className="loading"
      >
        <div>
          <img
            alt="loading"
            src="rolling.svg"
          />
          <p
            className="loading__text"
          >
            Loading...
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
